src/steamship/app/lambda_handler.py

```diff
@@ -159,9 +151,55 @@ def handler(event: Dict, context: Dict = None) -> dict:
             # The below should make it so calls to logging.info etc are also routed to the remote logger
             logging.root.addHandler(logging_handler)
 
-        response = _handler(event, context)
+        try:
+            client = Steamship(
+                **{to_snake_case(k): v for k, v in event.get("clientConfig", {}).items()}
+            )
+        except SteamshipError as se:
+            logging.exception(se)
+            return Response.from_obj(se).dict(by_alias=True)
+        except Exception as ex:
+            logging.exception(ex)
+            return Response.error(
+                code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                message="Plugin/App handler was unable to create Steamship client.",
+                exception=ex,
+            ).dict(by_alias=True)
+
+        response = _handler(event, client, context)
+
+        result = response.dict(by_alias=True)
+        # When created with data > 4MB, data is uploaded to a bucket.
+        # This is a very ugly way to get the deep size of this object
+        data_size = sys.getsizeof(json.dumps(result.get("data", {})))
+        logging.info(f"Response data size {data_size}")
+        if data_size > 4e6 and invocation_context.invocable_type == "plugin":
+            logging.info("Response data size >4MB, must upload to bucket")
+            # data_bucket: Optional[SignedUrl.Bucket] = None
+            # data_filepath: Optional[str] = None
+
+            filepath = str(uuid.uuid4())
+            signed_url_response = client.get_space().create_signed_url(
+                SignedUrl.Request(
+                    bucket=SignedUrl.Bucket.PLUGIN_DATA,
+                    filepath=filepath,
+                    operation=SignedUrl.Operation.WRITE,
+                )
+            )
+            signed_url = signed_url_response.data.signed_url
+            logging.info(f"Got signed url for writing: {signed_url}")
+
+            upload_to_signed_url(signed_url, json.dumps(result.get("data", None)).encode("UTF-8"))
+
+            # Now remove raw data and replace with bucket
+            result["data"] = None
+            result["dataBucket"] = str(SignedUrl.Bucket.PLUGIN_DATA.value)
```

Comment:
```suggestion
            result["dataBucket"] = SignedUrl.Bucket.PLUGIN_DATA.value
```

Isn't the value already a str?

Reply:
Gah, yes :)